great_expectations/core/validation_definition.py

```diff
@@ -221,6 +224,30 @@ def run(
         result_format: ResultFormat | dict = ResultFormat.SUMMARY,
         run_id: RunIdentifier | None = None,
     ) -> ExpectationSuiteValidationResult:
+        """
+        Runs a validation using the configured data and suite.
+
+        Args:
+            batch_parameters: The dictionary of parameters necessary for selecting the
+              correct batch to run the validation on. The keys are strings that are determined
+              by the BatchDefinition used to instantiate this ValidationDefinition. For example:
+
+              - whole table -> None
+              - yearly -> year
+              - monthly -> year, month
+              - daily -> year, month, day
+
+            expectation_parameters: A dictionary of parameters values for any expectations using
+              parameterized values (the $PARAMETER syntax). The keys are the parameter names
+              and the values are the values to be used for this validation run.
+            result_format: A parameter controlling how much diagnostic information the result
+              contains.
+            checkpoint_id: You likely want to leave this as None. This is inserted into
```

Comment:
> You likely want to leave this as None
Nit: the tone here feels kinda off to me. Should it be something more along the lines of `id of the checkpoint IFF this is being invoked via `checkpoint.run`. Worth noting that I think James wanted to push back on passing this in.

Reply:
Sounds good. I'll update this and the same statement for run_id.